src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -497,11 +497,20 @@ public async Task GetAsync_ContentCanBeCanceled(int getMode, int cancelMode)
                                 httpClient.CancelPendingRequests();
                                 break;
 
-                            // case 2: timeout fires on its own
+                                // case 2: timeout fires on its own
                         }
 
                         await tcs.Task;
                     });
+
+                    // The client may have completed before even sending a request when testing HttpClient.Timeout.
+                    await Task.WhenAny(serverHandling, tcs.Task);
+                    if (cancelMode != 2)
+                    {
+                        // If using a timeout to cancel requests, it's possible the server's processing could have gotten interrupted,
+                        // so we want to ignore any exceptions from the server when in that mode.  For anything else, let exceptions propagate.
+                        await serverHandling;
```

Comment:
should we wrap this in try/catch for cancelMode == 2 to make sure the Task is finished when test is done?

Reply:
> to make sure the Task is finished when test is done?

It may not be.  If the client gets canceled before it actually initiates the request to the server, the server will wait forever.